tot/plot_utils.py

```diff
@@ -0,0 +1,374 @@
+import logging
+
+import numpy as np
+import plotly.graph_objs as go
+from plotly_resampler import register_plotly_resampler, unregister_plotly_resampler
+
+from tot.df_utils import prep_or_copy_df
+
+log = logging.getLogger("tot.plot")
+
+
+# UI Configuration
+prediction_color = "#2d92ff"
+actual_color = "black"
+trend_color = "#B23B00"
+line_width = 2
+marker_size = 4
+xaxis_args = {
+    "showline": True,
+    "mirror": True,
+    "linewidth": 1.5,
+}
+yaxis_args = {
+    "showline": True,
+    "mirror": True,
+    "linewidth": 1.5,
+}
+layout_args = {
+    "autosize": True,
+    "template": "plotly_white",
+    "margin": go.layout.Margin(l=0, r=10, b=0, t=10, pad=0),
+    "font": dict(size=10),
+    "title": dict(font=dict(size=12)),
+    "hovermode": "x unified",
+}
+
+
+def log_warning_colab_resampler():
+    log.warning(
+        "Warning: plotly-resampler not supported for google colab environment. "
+        "Plotting backend automatically switched to 'plotly' without resampling "
+    )
+
+
+def log_warning_static_env_resampler():
+    log.warning(
+        "Warning: plotly-resampler not supported for this environments. "
+        "Plotting backend automatically switched to 'plotly' without resampling "
+    )
+
+
+def log_value_error_invalid_plotting_backend_input():
+    raise ValueError(
+        "Selected plotting backend invalid. Set plotting backend to one of the "
+        "valid options 'plotly','plotly-auto','plotly-resampler'."
+    )
+
+
+def log_value_error_invalid_highlight_forecast_input():
+    raise ValueError(
+        "input for highlight_forecast invalid. Set highlight_forecast step equal to"
+        " or smaller than the prediction horizon"
+    )
+
+
+def validate_current_env():
+    """
+    Validate the current environment to check if it is a valid environment to run the code.
+
+    Returns
+    -------
+    bool :
+        True if the current environment is a valid environment to run the code, False otherwise.
+
+    """
+    from IPython.core.getipython import get_ipython
+
+    if "google.colab" in str(get_ipython()):
+        log_warning_colab_resampler()
+        vaild_env = False
```

Comment:
You can directly return this value in any of these cases, might be easier to read and less nested code
```suggestion
        return False
```